### PR TITLE
blob/tests: add a conformance test for ListObject.As for a directory

### PR DESCRIPTION
**Pull request by vangent**

### CODE REVIEW

File: blob/s3blob/s3blob_test.go [hunk 1]
```diff
@@ -140,9 +140,16 @@ func (verifyContentLanguage) ReaderCheck(r *blob.Reader) error {
...
+		var commonPrefix s3.CommonPrefix
+		if !o.As(&commonPrefix) {
+			return errors.New("ListObject.As for directory returned false")
+		}
+		return nil
+	}
 	var obj s3.Object
 	if !o.As(&obj) {
-		return errors.New("ListObject.As returned false")
+		return errors.New("ListObject.As for blob returned false")
```
Comment (shantuo) [inline blob/s3blob/s3blob_test.go]: nit: s/blob/object

File: blob/azureblob/azureblob_test.go [hunk 2]
```diff
@@ -195,9 +195,16 @@ func (verifyContentLanguage) ReaderCheck(r *blob.Reader) error {
...
+		var prefix azblob.BlobPrefix
+		if !o.As(&prefix) {
+			return errors.New("ListObject.As for directory returned false")
+		}
+		return nil
+	}
 	var item azblob.BlobItem
 	if !o.As(&item) {
-		return errors.New("ListObject.As returned false")
+		return errors.New("ListObject.As for blob returned false")
```
Comment (shantuo) [inline blob/azureblob/azureblob_test.go]: nit: s/blob/object

File: blob/s3blob/s3blob_test.go (hunk 1 above)
Comment (vangent) [reply blob/s3blob/s3blob_test.go]: Done.

File: blob/azureblob/azureblob_test.go (hunk 2 above)
Comment (vangent) [reply blob/azureblob/azureblob_test.go]: Done.

**Pull request merged**